### PR TITLE
Add support for generating IdToken and UserInfo example payloads (#411)

### LINKED ISSUE #411: Add support for generating example IdToken and UserInfo for client-scope evaluation
### Description

The old admin-console supports to tests scope / claim mappings by allowing to generate id-token and userinfo payload examples for a given user.

### Discussion

_No response_

### Motivation

This feature enables users to test claim mappers for id token and user info responses.

### Details

Add methods like
- `evaluateGenerateIdToken`
- `evaluateGenerateUserInfo`

Next to the already existing `evaluateGenerateAccessToken`

### PR DESCRIPTION
**Pull request by thomasdarimont**
Fixes #411

See: https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/ClientScopeEvaluateResource.java#L159

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

### CODE REVIEW

**Comment by ssilvert** [comment]
Also related to https://github.com/keycloak/keycloak-admin-ui/issues/1564

File: src/resources/clients.ts [hunk 1]
```diff
@@ -411,6 +411,27 @@ export class Clients extends Resource<{realm?: string}> {
     queryParamKeys: ['scope', 'userId'],
   });
 
+  public evaluateGenerateIdToken = this.makeRequest<
+  {id: string; scope: string; userId: string},
+  object
```
Comment (jonkoops) [inline src/resources/clients.ts:416]: According to the source the return type for this one is `Map<String, Object>`, so we can use a more narrow type here in TypeScript as well.
```suggestion
  Record<string, unknown>
```

File: src/resources/clients.ts [hunk 2]
```diff
@@ -411,6 +411,27 @@ export class Clients extends Resource<{realm?: string}> {
...
+  >({
+    method: 'GET',
+    path: '/{id}/evaluate-scopes/generate-example-id-token',
+    urlParamKeys: ['id'],
+    queryParamKeys: ['scope', 'userId'],
+  });
+
+  public evaluateGenerateUserInfo = this.makeRequest<
+  {id: string; scope: string; userId: string},
+  object
```
Comment (jonkoops) [inline src/resources/clients.ts:426]: This method seems to return a type of [`IdToken`](https://www.keycloak.org/docs-api/16.1/rest-api/index.html#_idtoken), let's make this reflect that.

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Comment by thomasdarimont** [comment]
@jonkoops thanks for taking a look, how about returning Record<string, unknown> from both methods and add the more precise type information later?

I started to create the proper type definitions for JsonWebToken, IDToken, AccessToken and UserInfo - saved those in [this gist](https://gist.github.com/thomasdarimont/d8272edb303bda2d1ddfc93a26512a5a).
I was not sure whether this was necessary as there is already `AccessTokenRepresentation.ts`. Perhaps you can take it from there :)

**Comment by jonkoops** [comment]
@thomasdarimont could you give this PR a rebase? Our CI was borked for a while but should be good now.

**Comment by thomasdarimont** [comment]
@jonkoops thanks for the feedback. I just rebased and updated the PR. I noticed that we now already have a `evaluateGenerateIdToken` and `evaluateGenerateUserInfo` function.

The only effective change this PR brings is changing the return type from `object` to `Record<string, unknown>` which might require projects to change code.

Hence, I think this PR is obsolete now and can be closed.

**Pull request merged**